feat: Add docs about when to use the loader/npm (#8249)
getsentry/sentry-docs: src/platforms/javascript/common/install/index.mdx

---

<PageGrid />

> ## When to use which Installation Method

Reviewer: ```suggestion
## How To Decide Which Installation Method To Use
```

Depending on the concrete needs of your application, you may wonder which installation method you should use.

> ### Why to use the Loader Script over NPM

Reviewer 2: ```suggestion
### Reasons To Choose The Loader Script
```

Just a suggestion: Could also be "Why To Choose..." I think.

Reviewer 3: I like the suggestion above: "### Reasons To Choose The Loader Script"

If we don't accept the above suggestion, then I'd say "Why to use..." sounds funny. IMO it should be either "Why use..." or "When to use..."

Reviewer 3: ```suggestion
### When to use the Loader Script over NPM
```

Reviewer: "Why choose...over" makes more sense to me than "When to use...over"

> Using the Loader Script is the easiest way to start using Sentry. You only need to add a script tag to your application, and we'll take care of setting everything up correctly for you.

Reviewer: ```suggestion
Using the Loader Script is the easiest way to start using Sentry. Add a script tag to your application, and we'll take care of setting everything else up correctly for you.
```

- **Easy to set up**: You only need to [add a script tag](./loader/#using-the-loader) to your application.

> - **Easy to upgrade**: We'll take care of always shipping the latest version of the SDK to your users - no action needed on your side.

Reviewer 4: Should we talk about _why_ that's important/valuable? It's so you're ALWAYS up-to-date with the latest feature releases, bug fixes, improvements, etc.!

Reviewer: Plus one to that! It's always helpful to tell our users why they should do something :)

Author: Added this sentence:

> This ensures you'll always get the latest features and bug fixes of the Sentry SDK.

> 
> - **Easy to set up**: You only need to [add a script tag](./loader/#using-the-loader) to your application.
> 
> - **Easy to upgrade**: We'll take care of always shipping the latest version of the SDK to your users - no action needed on your side.

Reviewer: ```suggestion

- **Easy to set up**: Just [add a script tag](./loader/#using-the-loader) to your application.

- **Easy to upgrade**: We'll take care of always shipping the latest version of the SDK to your users. No action's needed from you.
```

> - **Easy to add additional configuration**: You can still configure Sentry to your liking via `Sentry.onLoad()`.

Reviewer 5: l: Should we link to the respective page?

Author: I added links! good point :)

> - **Easy to add additional configuration**: You can still [configure the SDK](./loader/#sdk-configuration) to your liking via `Sentry.onLoad()`.

Reviewer 4: An example of what configs someone might do? What's the most common one?

Author: I think since this is linked now this should be good enough, as you can find examples etc. there? I just wanted to point out that you can still configure stuff when using the loader :)

> - **Easy to add additional configuration**: You can still [configure the SDK](./loader/#sdk-configuration) to your liking via `Sentry.onLoad()`.

Reviewer: ```suggestion
- **Easy to add additional configuration**: You can [configure the SDK](./loader/#sdk-configuration) to your liking via `Sentry.onLoad()`.
```

> - **Lazy-loading for errors-only mode**: If you only want to use Sentry for errors, we'll only load the SDK when an error occurs. This means that we can reduce the amount of JavaScript loaded on your page until an error occurs.

Reviewer 2: ```suggestion
- **Lazy-loading**: If you only want to use Sentry for errors, we'll only load the SDK when an error occurs. This means that we can reduce the amount of JavaScript loaded on your page until an error occurs.
```

Don't know about this one a 100% but it's catchier (ie. if I scan for lazy load my eyes will find it more easily) and we mention that it's for errors only anyhow in the following text and the note.

> - **Lazy-loading**: If you only want to use Sentry for errors, we'll only load the SDK when an error occurs. This means that we can reduce the amount of JavaScript loaded on your page until an error occurs.

Reviewer 4: Should we say directly 'reduce bundle size'. That is what this is, right?

Author: I added this sentence:
> This reduces the initial bundle size.

> - **Lazy-loading**: If you only want to use Sentry for errors, we'll only load the SDK when an error occurs. This means that we can reduce the amount of JavaScript loaded on your page until an error occurs.

Reviewer: ```suggestion
- **Lazy-loading**: If you just want to use Sentry for errors, we'll only load the SDK when an error occurs, reducing the amount of JavaScript loaded on your page until then.
```

  <Note>

>     Note that when enabling Replay and/or Performance, the SDK will be loaded

Reviewer 2: ```suggestion
    Note that when enabling Session Replay and/or Performance functionality, the SDK will be loaded
```

    immediately because we need to capture what's happening on the page as early
    as possible.
  </Note>

> ### Why to use NPM over the Loader Script

Author: Hmm if we go with this naming we have to rewrite the bullet points here 🤔 Currently it is always relative to the loader, we'd have to rewrite this to e.g. `Use framework-specific initialization` or `Full public API`. WDYT, which is the better/more easily understandable way to phrase this?

> ### Why to use NPM over the Loader Script

Reviewer 3: If we want to change away from the original title, and don't want to rewrite the bullets, maybe this works better:

```suggestion
### When to change from the Loader Script to NPM Package
```

it's a longer sentence though

> ### Why to use NPM over the Loader Script

Author: I went with` Why Choose X over Y`, and slightly reworded the bullet points to be "positive" so this makes sense!

While using the Loader Script has definite advantages, it also comes with some drawbacks:

> - **No framework-specific initialization**: As of now, the Loader Script only serves the generic JavaScript Browser SDK. This means that you'll not get any React, Vue, or similar framework-specific initialization. If you need this, you'll need to install the framework SDK (e.g. `@sentry/react` or `@sentry/vue`) via npm.

Reviewer 2: ```suggestion
- **No framework-specific initialization**: As of now, the Loader Script only provides generic Browser JavaScript instrumentation. This means that you'll not get any React, Vue, or similar framework-specific features for Sentry. If you want this, you'll need to install the framework SDK (e.g. `@sentry/react` or `@sentry/vue`) via npm.
```

> - **Reduced public API**: The Loader Script only exposes a subset of public APIs (e.g. `Sentry.captureException()`, ...).

Reviewer 2: ```suggestion
- **Reduced public API**: The Loader Script only exposes a subset of public APIs (e.g. `Sentry.captureException()`, ...). This means if you're planning to have a lot of custom functionality, you're likely better off with the npm package.
```

> - **No control over the SDK version**: If you happen to need exact control over the SDK version, you'll need to install the SDK via npm.

Author: Yeah, I think this is OK for the purpose of this overview 😅 not 100% technically right but the point is I'd say :D

> - **No framework-specific initialization**: As of now, the Loader Script only provides generic Browser JavaScript instrumentation. This means that you'll not get any React, Vue, or similar framework-specific features for Sentry. If you want this, you'll need to install the framework SDK (e.g. `@sentry/react` or `@sentry/vue`) via npm.
> 
> - **Reduced public API**: The Loader Script only exposes a subset of public APIs (e.g. `Sentry.captureException()`, ...). This means if you're planning to have a lot of custom functionality, you're likely better off with the npm package.
> 

Reviewer: ```suggestion
- **No framework-specific initialization**: As of now, the Loader Script only provides generic Browser JavaScript instrumentation. This means that you won't get any React, Vue, or similar framework-specific features for Sentry. To get these features, you'll need to install the framework SDK (e.g. `@sentry/react` or `@sentry/vue`) via npm.

- **Reduced public API**: The Loader Script only exposes a subset of public APIs (for example, `Sentry.captureException()`, ...). If you need a lot of custom functionality, you're likely better off with the npm package.

```

> - **No control over the SDK version**: If you happen to need exact control over the SDK version, you'll need to install the SDK via npm.

Reviewer: ```suggestion
- **No control over the SDK version**: If you need full control over the SDK version, you'll need to install the SDK via npm.
```
